libsolidity/ast/ASTAnnotations.h

```diff
@@ -235,6 +235,7 @@ struct TryCatchClauseAnnotation: ASTAnnotation, ScopableAnnotation
 
 struct ForStatementAnnotation: StatementAnnotation, ScopableAnnotation
 {
+	util::SetOnce<bool> isSimpleCounterLoop;
```

Comment:
I think we should single those out into something like "optimizer hint annotations" to clearly separate them from type checker annotations.